src/Components/TableItem.js

```diff
@@ -0,0 +1,273 @@
+import React, { Component } from 'react';
+
+class TableItem extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      saveItem: {},
+      isEditing: false
+    }
+  }
+
+  // mount data before render
+  componentWillMount() {
+    this.setState({
+      saveItem: this.props.tableItem
+    });
+  }
+
+  static defaultProps = {
+    categories: ['Billing', 'Dev', 'Marketing', 'Service'],
+    assignees: ['Erwin', 'Jessica', 'George', 'Jose', 'Luke'],
+    priorities: ['Normal', 'Low', 'Medium', 'High'],
+    status: ['Open', 'Pending', 'Processing', 'Closed']
+  }
+  
+  // Edit: set saveItem to new tableItem
+  editItem = () => {
+    this.setState({
+      saveItem: this.props.tableItem,
+      isEditing: true
+    })
+  }
+
+  // get new value when editing
+  handleChange = () => {
+    this.setState({
+      saveItem: {
+        ...this.state.saveItem,
+        category: this.refs.category.value,
+        subject: this.refs.subject.value,
+        assignee: this.refs.assignee.value,
+        priority: this.refs.priority.value,
+        status: this.refs.status.value
+      }
+    });
+  }
+
+  // Save: set update to true and props saveItem, then set is editing to false
+  saveItem = (item) => {
+    if (this.refs.subject.value === '') {
+      alert('Subject can\'t be empty');
+    } else {
+      this.setState({
+        saveItem: {
+          ...this.state.saveItem,
+          update: true
+        }
+      }, function () {
+        this.props.saveItem(this.state.saveItem);
+        this.setState({
+          isEditing: false
+        })
+      });
+    }
+  }
+
+  // Cancel: reset saveItem to original props data
+  cancelEdit = () => {
+    this.setState({
+      isEditing: false,
+      saveItem: this.props.tableItem
+    });
+  }
+
+  // Delete: props tableItem.id
+  deleteItem = (id) => {
+    id = this.props.tableItem.id;
+    this.props.onDelete(id);
+  }
+
+  // styling: set different badge color
+  handlePriorityColor(priority) {
+    if (priority === 'Low') {
+      return 'badge badge-pill badge-success'
+    } else if (priority === 'Medium') {
+      return 'badge badge-pill badge-warning'
+    } else if (priority === "High") {
+      return 'badge badge-pill badge-danger'
+    } else {
+      return 'badge badge-pill badge-secondary'
+    }
+  }
```

Comment:
Recommend you to use [classnames](https://github.com/JedWatson/classnames) to handle those logic